miso-web/src/main/webapp/scripts/pane_barcode_search.js

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2012. The Genome Analysis Centre, Norwich, UK
+ * MISO project contacts: Robert Davey @ TGAC
+ * *********************************************************************
+ *
+ * This file is part of MISO.
+ *
+ * MISO is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MISO is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MISO.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * *********************************************************************
+ */
+
+PaneTarget.barcode_search = (function() {
+  var url = "/miso/rest/barcodable/search";
+
+  var transform = function(searchResult) {
+    return Tile.make([
+        Tile.title(searchResult.entityType),
+        Tile.lines(["Name: " + searchResult.name, "Alias: " + searchResult.alias, "Barcode: " + searchResult.identificationBarcode])],
+      function() {
+        // todo: click handler
+        console.log(window.location.origin);
+        console.log(searchResult.url);
+        window.location = window.location.origin + searchResult.url;
+      }
+    );
+  };
+
+  var search = function(query, onLoad, onSuccess, onError) {
+    onLoad();
+
+    jQuery.ajax({
+      'dataType': 'json',
+      'type': 'GET',
+      'url': url + "/?q=" + query,
```

Comment:
Maybe encoding the query in the URL would be a better idea?

For example, if a "" character is sent, MISO will respond with "Bad Request".

Reply:
Can use `jQuery.param({q: query})`

Reply:
Yes. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI